File: DomainModels/Models/Room.cs
using System.Text.Json.Serialization;

//Class to handle rooms
namespace DomainModels.Models.Entities
{
    public class Room
    { 
        public int Id { get; set; }
        public int Capacity { get; set; }
        public RoomType RoomType { get; set; }
        public required int RoomNumber { get; set; }
        public decimal PricePerNight { get; set; }
        public List<string>? Facilities { get; set; }
        // Navigation property to Booking

        //JsonIgnore to avoid errors in post requests for rooms and booking.
        [JsonIgnore]
        public List<Booking>? Availabilities { get; set; } 
    }
}

File: React/admin-panel/src/interfaces/room.ts
import { RoomType } from "./roomtype";

// Interface to handle Room details in typescript only.
export interface Room {
    id: number;
    capacity: number;
    roomType: RoomType;
    roomNumber: number;
    pricePerNight: number;
    facilities: string[] | [] | null; //Nullable since there are no set facilities
  }

  //Interface to create rooms in typescript
  export interface CreateRoomDTO {
    capacity: number;
    roomTypeId: number;
    roomNumber: number;
    pricePerNight: number;
    facilities: string[] | [] | null; //Nullable since there are no set facilities
  }